service/src/main/java/uk/nhs/adaptors/gp2gp/common/task/TaskExecutor.java

```diff
@@ -0,0 +1,8 @@
+package uk.nhs.adaptors.gp2gp.common.task;
+
+public interface TaskExecutor {
+
+    Class<? extends TaskDefinition> getTaskType();
```

Comment:
Change to generic type T

Reply:
Still needs doing

public interface TaskExecutor<T extends TaskDefinition> {

    Class<T> getTaskType();

    void execute(T taskDefinition);
}